jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/FileMappedHttpContentFactory.java

```diff
@@ -0,0 +1,110 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.thread.AutoLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FileMappedHttpContentFactory implements HttpContent.Factory
+{
+    private static final Logger LOG = LoggerFactory.getLogger(FileMappedHttpContentFactory.class);
+
+    private final HttpContent.Factory _factory;
+    private final int _minFileSize;
+
+    public FileMappedHttpContentFactory(HttpContent.Factory factory)
+    {
+        this(factory, 16 * 1024);
+    }
+
+    public FileMappedHttpContentFactory(HttpContent.Factory factory, int minFileSize)
+    {
+        _factory = Objects.requireNonNull(factory);
+        _minFileSize = minFileSize;
+    }
+
+    @Override
+    public HttpContent getContent(String path) throws IOException
+    {
+        HttpContent content = _factory.getContent(path);
+        if (content != null && content.getContentLengthValue() > _minFileSize)
+            return new FileMappedContent(content);
+        return content;
+    }
+
+    public static class FileMappedContent extends HttpContentWrapper
```

Comment:
This should be `private` or maybe package-protected.

Reply:
I made this private now